broadinstitute/single_cell_portal_core
# Restore split/merge functionality (SCP-4728)

To see the code changes for functionality check out the first commit. The second commit is only renaming `splitArrayLabels` and `setSplitArrayLabels` to `isSplitArrayLabels` and `updateIsSplitArrayLabels`.

This restores the functionality of merge/split array labels. 


https://user-images.githubusercontent.com/54322292/199084080-1e620263-ff2a-4623-91cf-700305d1cde8.mov
*Edit - this clip shows persistence of hidden traces between toggles but please note this was the first iteration of this PR and that persistence between split/merge will NOT be supported at this time.


To test
- You need a study that has split labels. I used the large study from staging <study> copied to my local and because of the bug with overrides hiding legend search and showing the split/merge button this worked haha. It might be simpler to use a smaller study so you don't need to worry about that at all but I don't know of one with the splits.
- Go to the explore tab for the study
- click merge or split
- then click it again to see that the legend traces do update appropriately
